components/kyma-environment-broker/internal/orchestration/manager/manager.go

```diff
@@ -67,14 +71,9 @@ func (m *orchestrationManager) Execute(orchestrationID string) (time.Duration, e
 		return m.failOrchestration(o, fmt.Errorf("failed to get orchestration: %w", err))
 	}
 
-	maintenancePolicy, err := m.getMaintenancePolicy()
-	if err != nil {
-		m.log.Warnf("while getting maintenance policy: %s", err)
-	}
-
-	operations, err := m.resolveOperations(o, maintenancePolicy)
+	operations, err := m.waitForStart(o)
 	if err != nil {
-		return m.failOrchestration(o, fmt.Errorf("failed to resolve operations: %w", err))
+		return 0, err
```

Comment:
Shouldn't we just log the error, and  allow retrying the `waitForStart` in temporary error?

```suggestion
		return m.pollingInterval, nil
```

Reply:
I kept `return m.failOrchestration(o, fmt.Errorf("failed to resolve operations: %w", err))` instead. A failOrchestration() needs be to triggered here to retry failed orchestration. Also when orchestration is empty, failOrchestration() is needed so that o.IsFinished() can quit orchestration.